Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Models/TypeModelBuilder.cs

```diff
@@ -11,14 +11,32 @@ public static class TypeModelBuilder
     {
         public static TypeModel Build(ITypeSymbol symbol, GeneratorExecutionContext context)
         {
+            bool isGenericType;
+            if (symbol is INamedTypeSymbol)
+                isGenericType = ((INamedTypeSymbol)symbol)?.IsGenericType ?? false;
+            else
+                isGenericType = false;
+
+            IList<TypeModel> typeArguments;
+            if (symbol is INamedTypeSymbol)
+                typeArguments = ((INamedTypeSymbol)symbol)?.TypeArguments.Select(arg => Build(arg, context)).ToList();
+            else
+                typeArguments = new List<TypeModel>();
+
+            bool isEnumerable;
+            if (symbol is INamedTypeSymbol)
+                isEnumerable = ((INamedTypeSymbol)symbol)?.Interfaces.Any(it => it.Equals(context.Compilation.GetTypeByMetadataName(TypeFullNames.IEnumerable), SymbolEqualityComparer.Default)) ?? false;
+            else
```

Comment:
Lets be defensive lets make this an `else if(symbol is IDynamicTypeSymbol)` and then I think the final `else` block would set `isEnumerable` to `false`.

Reply:
Fixed